reactive/flannel.py

```diff
@@ -1,254 +1,224 @@
-              stdout=PIPE,
-              stderr=STDOUT,
-              close_fds=True)
-    version = p.stdout.read()
-    if version:
-        hookenv.application_version_set(version.split(b'v')[-1].rstrip())
-
-
-@when('flannel.sdn.configured', 'host.connected')
-@when_not('flannel.host.relayed')
-def relay_sdn_configuration(plugin_host):
-    ''' send the flannel interface configuration to the principal unit '''
-    try:
-        subnet, mtu = _ingest_network_config()
-        cidr = hookenv.config('cidr')
-        plugin_host.set_configuration(mtu, subnet, cidr)
-        set_state('flannel.host.relayed')
-        hookenv.status_set('active', 'Flannel subnet {0}'.format(subnet))
-    except TypeError:
-        hookenv.status_set('waiting', 'Flannel is starting up.')
+    status_set('blocked', 'Waiting for etcd relation.')
 
 
 @hook('upgrade-charm')
 def reset_states_and_redeploy():
     ''' Remove state and redeploy '''
-    remove_state('flannel.host.relayed')
-    remove_state('flannel.installed')
+    remove_state('flannel.binaries.installed')
+    remove_state('flannel.service.started')
+    remove_state('flannel.version.set')
+    remove_state('flannel.network.configured')
+    remove_state('flannel.service.installed')
 
 
 @hook('stop')
 def cleanup_deployment():
     ''' Terminate services, and remove the deployed bins '''
-    host.service_stop('flannel')
-
-    # Stop and remove the flannel bridge
-    down = ['ip', 'link', 'set', 'flannel.1', 'down']
-    delete = ['ip', 'link', 'delete', 'flannel.1']
+    service_stop('flannel')
+    down = 'ip link set flannel.1 down'
+    delete = 'ip link delete flannel.1'
     try:
-        subprocess.check_call(down)
-        subprocess.check_call(delete)
-    except subprocess.CalledProcessError:
-        # We failed to stop or remove the flannel interface
-        # Its difficult to discern when this should halt execution... for now
-        # just allow it to fail and leave the interface up. More data will
-        # need to be collected to resolve this particular code path, as it
-        # worked when testing.
-        hookenv.log('Unable to remove iface flannel.1')
-        hookenv.log('Potential indication that cleanup is not possible')
-
-    # List of files we expect to need to clean up
+        check_call(split(down))
+        check_call(split(delete))
+    except CalledProcessError:
+        log('Unable to remove iface flannel.1')
+        log('Potential indication that cleanup is not possible')
     files = ['/usr/local/bin/flanneld',
              '/lib/systemd/system/flannel',
              '/lib/systemd/system/flannel.service',
-             '/etc/ssl/flannel/client-ca.pem',
-             '/etc/ssl/flannel/client-cert.pem',
-             '/etc/ssl/flannel/client-key.pem',
-             '/var/run/flannel/subnet.env']
-    # if the file exists on disk, remove it to self-cleanup
+             '/var/run/flannel/subnet.env',
+             '/usr/local/bin/flanneld',
```

Comment:
The install code also puts etcdtl in `/usr/local/bin/`. I suppose that is not a service and may not need to get cleaned up.